src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const array = [...sourceString.split(';')];
+
+  array.map((element, i) => [
+    element.slice(0, element.indexOf(':') + 1),
+    element.slice(element.indexOf(':') + 1, -1),
+  ]);
+
+  array.map((element) => element[0] + element[1]);
+
+  const newObject = { ...array };
```

Comment:
The spread operator is incorrectly used here. You cannot spread an array directly into an object like this. Instead, consider using `Object.fromEntries()` with an array of key-value pairs to create the object.